### DISCUSSION TITLE
CPU overlock settings

### DISCUSSION THREAD

**Discussion by StormedBubbles** [StormedBubbles/mame2010-libretro#2] [Q&A]
To anyone who may come across this,

Does anyone know how to enable the CPU clock settings? Ideally, I'd like to have that adjustable in the core options so that per-game settings can be saved. I think that underclocking could greatly benefit the sound in Ghoul Panic and the two Golgo 13 games when played on a Raspberry Pi 4B. Those games are so close to "perfect" performance!

@pjft sorry for the ping, but I came across [some discussion on the RetroPie forums](https://retropie.org.uk/forum/topic/26807/deathsmiles-on-lr-mame2015-any-performance-tips/) and saw the work you did for the lr-mame2016 core's CPU options. I added lightgun support to my fork of this core (see announcement post for credit and instructions) so that a handful of games otherwise unplayable can be played on a Pi with Sinden lightguns. Do you happen to have any insights on getting a core option for CPU clock in lr-mame2010?

**Comment by pjft** [comment]
Well, what I did was turn an existing lr-mame2016 capability into a core option. I did not implement CPU clock for lr-mame2016.

I'd very much follow [the original PR I submitted for lr-mame2016](https://github.com/libretro/mame2016-libretro/pull/43) but chances are that the actual ability to change CPU clock isn't present in lr-mame2010 so if that's the case unless you were to actually implement CPU clock settings there (if at all possible in that architecture?), we'd be out of luck.

From reading the discussion, it seems that those features were introduced in 2014 or something.

I'd start with accessing the tab menu and seeing if there's one such option and then go from there.

Best.
